### PR TITLE
tvOS Split View Presentation Support

### PR DESCRIPTION
**Pull request by biozal**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
allows tvOS to use Split View navigation in the presentation framework and allows annotations on all classes (master, detail).

### :arrow_heading_down: What is the current behavior?
Split View is not supported

### :new: What is the new behavior (if this is a feature change)?
Split View is supported

### :boom: Does this PR introduce a breaking change?
No - but the previous change to the new presentation framework did.

### :bug: Recommendations for testing
Use Playground tvOS application to test.

### :thinking: Checklist before submitting

- [X ] All projects build
- [X] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ ] Nuspec files were updated (when applicable)
- [X] Rebased onto current develop

### CODE REVIEW

**Comment by biozal** [comment]
I will be updating the documentation on Friday for this.  Currently there is no documentation for new presentation framework outside of the playground sample app.

File: MvvmCross/tvOS/tvOS/Views/MvxSplitViewController.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            ViewModel?.ViewDisappearing();
+        }
+
+        public override void ViewDidDisappear(bool animated)
+        {
+            base.ViewDidDisappear(animated);
+            ViewModel?.ViewDisappeared();
+        }
+
+        public virtual void ShowDetailView(UIViewController viewController, bool wrapInNavigationController)
```
Comment (martijn00) [inline MvvmCross/tvOS/tvOS/Views/MvxSplitViewController.cs]: I think we should move this into the presenter code.

File: MvvmCross/tvOS/tvOS/Views/MvxSplitViewController.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        {
+            viewController = wrapInNavigationController ? 
+                new MvxNavigationController(viewController) : viewController;
+
+            ShowDetailViewController(viewController, this);
+        }
+
+        public virtual void ShowMasterView(UIViewController viewController, bool wrapInNavigationController)
```
Comment (martijn00) [inline MvvmCross/tvOS/tvOS/Views/MvxSplitViewController.cs]: I think we should move this into the presenter code.

File: MvvmCross/tvOS/tvOS/Views/MvxSplitViewController.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            }
+        }
+
+        public override void PrepareForSegue(UIStoryboardSegue segue, NSObject sender)
+        {
+            base.PrepareForSegue(segue, sender);
+            this.ViewModelRequestForSegue(segue, sender);
+        }
+
+        public virtual bool CloseChildViewModel(IMvxViewModel viewModel)
```
Comment (martijn00) [inline MvvmCross/tvOS/tvOS/Views/MvxSplitViewController.cs]: I think we should move this into the presenter code.

File: MvvmCross/tvOS/tvOS/Views/IMvxSplitViewController.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using UIKit;
+using MvvmCross.Core.ViewModels;
+
+namespace MvvmCross.tvOS.Views
+{
+    public interface IMvxSplitViewController
```
Comment (martijn00) [inline MvvmCross/tvOS/tvOS/Views/IMvxSplitViewController.cs]: I think all of this could be controlled from the presenter?

**Comment by biozal** [comment]
I'll look at moving this code around tomorrow night

**Comment by biozal** [comment]
updated the code tonight per @martijn00 request.

**Comment by biozal** [comment]
added documentation for tvOS.

File: MvvmCross/tvOS/tvOS/Views/IMvxSplitViewController.cs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+﻿using UIKit;
+using MvvmCross.Core.ViewModels;
+
+namespace MvvmCross.tvOS.Views
+{
+    public interface IMvxSplitViewController : IMvxTvosView { }
```
Comment (martijn00) [inline MvvmCross/tvOS/tvOS/Views/IMvxSplitViewController.cs]: Can we remove this file?

File: MvvmCross/tvOS/tvOS/Views/MvxSplitViewController.cs [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+using MvvmCross.Binding.BindingContext;
+using MvvmCross.Core.ViewModels;
+
+namespace MvvmCross.tvOS.Views
+{
+    public class MvxSplitViewController 
+        : MvxEventSourceSplitViewController
+        ,IMvxTvosView
+        ,IMvxSplitViewController
```
Comment (martijn00) [inline MvvmCross/tvOS/tvOS/Views/MvxSplitViewController.cs]: This does not need to implement `IMvxSplitViewController`

File: MvvmCross/tvOS/tvOS/Views/Presenters/Attributes/MvxDetailSplitViewPresentationAttribute.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿using MvvmCross.Core.Views;
+namespace MvvmCross.tvOS.Views.Presenters.Attributes
+{
+    public class MvxDetailSplitViewPresentationAttribute : MvxBasePresentationAttribute
```
Comment (martijn00) [inline MvvmCross/tvOS/tvOS/Views/Presenters/Attributes/MvxDetailSplitViewPresentationAttribute.cs]: Please merge this into one, like https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross-Forms/MvvmCross.Forms/Views/Attributes/MvxMasterDetailPagePresentationAttribute.cs#L24

**Pull request merged**